src/tests/unittests/measurements/test_scans.py

```diff
@@ -364,6 +362,69 @@ def test_convert_scanjob_vec_scan1Dfast_adjust_sweeplength(self):
         self.assertEqual(sweepvalues._value_snapshot[0]['num'], 5)
         gates.close()
 
+    def test_convert_scanjob_vec_scan1Dfast_adjust_sweeplength_adjusted_end(self):
+        p = Parameter('p', set_cmd=None)
+        gates = VirtualIVVI(
+            name=qtt.measurements.scans.instrumentName('gates'), model=None)
+        station = qcodes.Station(gates)
+        station.gates = gates
+
+        scanjob = scanjob_t({'scantype': 'scan1Dfast',
+                             'sweepdata': {'param': p, 'start': -20, 'end': 20, 'step': .0075}})
+        _, sweepvalues = scanjob._convert_scanjob_vec(station)
+        actual_values = sweepvalues._values
+        self.assertEqual(actual_values[0], -20.0)
+        self.assertAlmostEqual(scanjob['sweepdata']['end'], 20.0 - 0.0025, 10)
+        self.assertEqual(sweepvalues._value_snapshot[0]['num'], 5334)
+
+        scanjob = scanjob_t({'scantype': 'scan1Dfast',
+                             'sweepdata': dict({'param': p, 'start': -500, 'end': 1,
+                                                'step': .8, 'wait_time': 3e-3})})
+        _, sweepvalues = scanjob._convert_scanjob_vec(station)
+        actual_values = sweepvalues._values
+        self.assertEqual(actual_values[0], -500.0)
+        self.assertAlmostEqual(scanjob['sweepdata']['end'], 1 - 0.2, 10)
+        self.assertEqual(sweepvalues._value_snapshot[0]['num'], 627)
+
+        gates.close()
+
+    def test_convert_scanjob_vec_end_equals_start(self):
+        p = Parameter('p', set_cmd=None)
+        gates = VirtualIVVI(
+            name=qtt.measurements.scans.instrumentName('gates'), model=None)
+        station = qcodes.Station(gates)
+        station.gates = gates
+
+        scanjob = scanjob_t({'scantype': 'scan1Dfast',
+                             'sweepdata': {'param': p, 'start': 20, 'end': 20, 'step': .0075}})
+        _, sweepvalues = scanjob._convert_scanjob_vec(station)
+        actual_values = sweepvalues._values
```

Comment:
```suggestion
        actual_values = list(sweepvalues)
```
Do not use the private `_values`. Altenatively, use `np.array(sweepvalues)`

Reply:
This is a nice solution. I will change this.